core/src/Kokkos_ExecPolicy.hpp

```diff
@@ -139,7 +138,23 @@ class RangePolicy : public Impl::PolicyTraits<Properties...> {
     check_conversion_safety(work_end);
     check_bounds_validity();
     set_auto_chunk_size();
-    set(args...);
+  }
+
+  template <typename IndexType1, typename IndexType2,
+            std::enable_if_t<(std::is_convertible_v<IndexType1, member_type> &&
+                              std::is_convertible_v<IndexType2, member_type>),
+                             bool> = false>
+  inline RangePolicy(const typename traits::execution_space& work_space,
```

Comment:
While we are at it how about we drop these pointless `inline` specifiers (here and in the previous constructor)?

Reply:
It's all over the file, and I didn't want to do it piecemeal.  How much clean up should I do here?  (It is simple to get rid of them all.)

Reply:
I would do the 2 constructors in this PR